src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,126 @@
+import { memo, useEffect, useMemo, useRef, useState } from 'react';
+import React from 'react';
+import debounce from 'lodash.debounce';
+import './Autocomplete.scss';
+import { peopleFromServer } from '../data/people';
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+
+type Props = {
+  delay?: number;
+  width?: number;
+  onSelected: (Person: Person | '') => void;
+  onWarning: (a: boolean) => void;
+};
+
+export const Autocomplete: React.FC<Props> = memo(
+  ({ delay = 300, width = 300, onSelected, onWarning }) => {
+    const [query, setQuery] = useState('');
+    const [appliedQuery, setAppliedQuery] = useState('');
+    const [showDrop, setShowDrop] = useState(true);
+    const [isFirstRender, setIsFirstRender] = useState(true);
+
+    const debouncedApplyQuery = useRef(
+      debounce((value: string) => {
+        setAppliedQuery(value);
+      }, delay),
+    ).current;
+
+    useEffect(() => {
+      return () => {
+        debouncedApplyQuery.cancel();
+      };
+    }, [debouncedApplyQuery]);
+
+    const filteredPeople = useMemo(() => {
+      return peopleFromServer.filter(person =>
+        person.name
+          .toLocaleLowerCase()
+          .includes(appliedQuery.toLocaleLowerCase()),
+      );
+    }, [appliedQuery]);
+
+    useEffect(() => {
+      if (!isFirstRender) {
+        if (filteredPeople.length === 0) {
```

Comment:
```suggestion
        if (!filteredPeople.length) {
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**